File: net/JetBrains.Serialization/src/Dataholders/AttributeInfos/UnknownAttributeInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Attribute = Org.BouncyCastle.Asn1.Cms.Attribute;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.Fields)]
public class UnknownAttributeInfo : AttributeInfo
{
    public override TextualInfo Identifier { get; }
    public IEncodableInfo Content { get; }

    private UnknownAttributeInfo(TextualInfo identifier, IEncodableInfo content)
    {
        Identifier = identifier;
        Content = content;
    }

    public UnknownAttributeInfo(Attribute attribute)
        : this(TextualInfo.GetInstance(attribute.AttrType), attribute.AttrValues.ToEncodableInfo())
    {
    }

    public override Asn1Encodable GetPrimitiveContent() => Content.ToPrimitive();
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/attributeInfos/UnknownAttributeInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import com.jetbrains.signatureverifier.serialization.toEncodableInfo
import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.cms.Attribute

@Serializable
data class UnknownAttributeInfo(
  override val identifier: TextualInfo,
  val content: EncodableInfo
) : AttributeInfo() {

  constructor(attribute: Attribute) : this(
    TextualInfo.getInstance(attribute.attrType),
    attribute.attrValues.toEncodableInfo()
  )

  override fun getPrimitiveContent() = content.toPrimitive()
}